src/custom_nodes/east_ocr/east_ocr.cpp

```diff
@@ -0,0 +1,399 @@
+    NODE_ASSERT((originalImageWidth % 4) == 0, "original image width must be divisible by 4");
+    std::string imageLayout = get_string_parameter("image_layout", params, paramsCount, "NCHW");
+    NODE_ASSERT(imageLayout == "NCHW" || imageLayout == "NHWC", "image layout can be either NCHW or NHWC for both inputs and outputs");
+
+    *infoCount = 3;
+    *info = (struct CustomNodeTensorInfo*)malloc(*infoCount * sizeof(struct CustomNodeTensorInfo));
+
+    (*info)[0].name = IMAGE_TENSOR_NAME;
+    (*info)[0].dimsCount = 4;
+    (*info)[0].dims = (uint64_t*)malloc((*info)->dimsCount * sizeof(uint64_t));
+    (*info)[0].dims[0] = 1;
+    if (imageLayout == "NCHW") {
+        (*info)[0].dims[1] = 3;
+        (*info)[0].dims[2] = originalImageHeight;
+        (*info)[0].dims[3] = originalImageWidth;
+    } else {
+        (*info)[0].dims[1] = originalImageHeight;
+        (*info)[0].dims[2] = originalImageWidth;
+        (*info)[0].dims[3] = 3;
+    }
+    (*info)[0].precision = FP32;
+
+    (*info)[1].name = SCORES_TENSOR_NAME;
+    (*info)[1].dimsCount = 4;
+    (*info)[1].dims = (uint64_t*)malloc((*info)->dimsCount * sizeof(uint64_t));
+    (*info)[1].dims[0] = 1;
+    (*info)[1].dims[1] = 1;
+    (*info)[1].dims[2] = originalImageHeight / 4;
+    (*info)[1].dims[3] = originalImageWidth / 4;
+    (*info)[1].precision = FP32;
+
+    (*info)[2].name = GEOMETRY_TENSOR_NAME;
+    (*info)[2].dimsCount = 4;
+    (*info)[2].dims = (uint64_t*)malloc((*info)->dimsCount * sizeof(uint64_t));
+    (*info)[2].dims[0] = 1;
+    (*info)[2].dims[1] = 5;
+    (*info)[2].dims[2] = originalImageHeight / 4;
+    (*info)[2].dims[3] = originalImageWidth / 4;
+    (*info)[2].precision = FP32;
+    return 0;
+}
+
+int getOutputsInfo(struct CustomNodeTensorInfo** info, int* infoCount, const struct CustomNodeParam* params, int paramsCount) {
+    int targetImageHeight = get_int_parameter("target_image_height", params, paramsCount, -1);
+    int targetImageWidth = get_int_parameter("target_image_width", params, paramsCount, -1);
+    NODE_ASSERT(targetImageHeight > 0, "original image height must be larger than 0");
+    NODE_ASSERT(targetImageWidth > 0, "original image width must be larger than 0");
+    bool convertToGrayScale = get_string_parameter("convert_to_gray_scale", params, paramsCount) == "true";
+    std::string imageLayout = get_string_parameter("image_layout", params, paramsCount, "NCHW");
+    NODE_ASSERT(imageLayout == "NCHW" || imageLayout == "NHWC", "image layout can be either NCHW or NHWC for both inputs and outputs");
+
+    *infoCount = 3;
+    *info = (struct CustomNodeTensorInfo*)malloc(*infoCount * sizeof(struct CustomNodeTensorInfo));
+
+    (*info)[0].name = TEXT_IMAGES_TENSOR_NAME;
+    (*info)[0].dimsCount = 5;
+    (*info)[0].dims = (uint64_t*)malloc((*info)->dimsCount * sizeof(uint64_t));
+    (*info)[0].dims[0] = 1;
+    (*info)[0].dims[1] = 0;
+    if (imageLayout == "NCHW") {
+        (*info)[0].dims[2] = convertToGrayScale ? 1 : 3;
```

Comment:
Do we have/will we have test for both using grayscale & not?

Reply:
We have no crnn model with grayscale I know of. Is there a way to convert it? @dtrawins